### ISSUE TITLE
The understanding of MSAA module

### ISSUE THREAD

**Issue by ouyangziyao** [XiaoBuL/CM-UNet#15]
Thank you for your excellent work. I'm a a beginner. The weight of **spatial_attention** multiple  **x_fused_s** and the add of **x_fused_s** and **x_fused_c** are I can't understand. As per **the picture**, I think it should be the **'x_fused_s'** multiple **' self.channel_attention(x_fused)'** and the **' torch.cat([x1, x2, x4], dim=1)'** add the result of  the **'x_fused_s'** multiple  **'self.channel_attention(x_fused)'**. They are different from your code. Furthemore, in the picture, the starting C1 and the dim of channel is the same. But, the channel dim in channel attention module is the dim after downsample. I think it's should be C2.
```
class FusionConv(nn.Module):
    def __init__(self, in_channels, out_channels, factor=4.0):
        super(FusionConv, self).__init__()
        dim = int(out_channels // factor)
        self.down = nn.Conv2d(in_channels, dim, kernel_size=1, stride=1)
        self.conv_3x3 = nn.Conv2d(dim, dim, kernel_size=3, stride=1, padding=1)
        self.conv_5x5 = nn.Conv2d(dim, dim, kernel_size=5, stride=1, padding=2)
        self.conv_7x7 = nn.Conv2d(dim, dim, kernel_size=7, stride=1, padding=3)
        self.spatial_attention = SpatialAttentionModule()
        self.channel_attention = ChannelAttentionModule(dim)
        self.up = nn.Conv2d(dim, out_channels, kernel_size=1, stride=1)
        self.down_2 = nn.Conv2d(in_channels, dim, kernel_size=1, stride=1)

    def forward(self, x1, x2, x4):
        
        x_fused = torch.cat([x1, x2, x4], dim=1)
        x_fused = self.down(x_fused)
        x_fused_c = x_fused * self.channel_attention(x_fused)
        x_3x3 = self.conv_3x3(x_fused)
        x_5x5 = self.conv_5x5(x_fused)
        x_7x7 = self.conv_7x7(x_fused)
        x_fused_s = x_3x3 + x_5x5 + x_7x7
        x_fused_s = x_fused_s * self.spatial_attention(x_fused_s)

        x_out = self.up(x_fused_s + x_fused_c)

        return x_out
```
![image](https://github.com/user-attachments/assets/d4b68f91-6668-4fa4-97fa-0de0d6f27d52)

**Comment by fcchit** [comment]
I found the same problem, and the code is not actually the same as described in the paper.

**Comment by CrazyBrick** [comment]
the same, can't understand....

![Image](https://github.com/user-attachments/assets/b86f7ce8-b2e7-45f9-a7f3-7e1bbe4a2017)

![Image](https://github.com/user-attachments/assets/182dd064-2efd-47a4-9b07-51a7fabc9499)

**Comment by MetroTY** [comment]
same problem...
